### PR TITLE
Feature 2194/Cacheable Role Permissions Lookup

### PR DESCRIPTION
**Pull request by ZacharyKlein**
- Lookup permissions by role
- Cache permissions per role
- Invalidate permissions cache on change

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/role/role_permissions/RolePermissionServicesImpl.java [hunk 1]
```diff
@@ -70,6 +76,13 @@ public List<RolePermission> findByRoleId(UUID roleId) {
...
 
+
+    @Cacheable
+    @Override
+    public List<RolePermission> findByRole(String role) {
+        return rolePermissionRepository.findByRole(role);
+    }
+
     @Override
     public List<Permission> findUserPermissions(@NotBlank UUID id) {
```
Comment (mkimberlin) [inline server/src/main/java/com/objectcomputing/checkins/services/role/role_permissions/RolePermissionServicesImpl.java:87]: Any particular reason not to make this method `@Cacheable` as well?

File: server/src/main/resources/application.yml [hunk 2]
```diff
@@ -14,6 +14,8 @@ micronaut:
       expire-after-write: 300
       heap:
         max-entries: 600
+    role-permission-cache:
+      expire-after-write: 86400
```
Comment (mkimberlin) [inline server/src/main/resources/application.yml:18]: This is pretty long...but, I guess why not? The service invalidates if we change things, so I guess I'm down. 🤷

Comment (ZacharyKlein) [reply server/src/main/resources/application.yml:18]: I think I just used the same value as the member cache.

File: server/src/main/java/com/objectcomputing/checkins/services/role/role_permissions/RolePermissionServicesImpl.java (hunk 1 above)
Comment (ZacharyKlein) [reply server/src/main/java/com/objectcomputing/checkins/services/role/role_permissions/RolePermissionServicesImpl.java:87]: Probably not! It's not the one being used in the annotation lookup, but it is used during login.

**Review by mkimberlin — APPROVED** [review]

**Review by mjperry91 — APPROVED** [review]

**Pull request merged**